tests/test_statelib_state.py

```
@@ -0,0 +1,219 @@
+
+def test_combine_partitions_overlap_raises() -> None:
+    state: sl.FlatState[int] = sl.FlatState.from_pytree({"a": 1, "b": 2})
+    first, second = sl.partition_state(state, keys={("a",)})
+
+    with pytest.raises(ValueError):
+        sl.combine_partitions(first, first)
+
+
+def test_combine_partitions_mismatched_sources_raises() -> None:
+    state: sl.FlatState[int] = sl.FlatState.from_pytree({"a": 1, "b": 2})
+    first, second = sl.partition_state(state, keys={("a",)})
```

Ruff (0.14.0)

Unpacked variable `second` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)